server/src/main/java/org/opensearch/bootstrap/Security.java

```diff
@@ -416,6 +422,29 @@ private static void addSocketPermissionForHttp(final Permissions policy, final S
         addSocketPermissionForPortRange(policy, httpRange);
     }
 
+    /**
+     * Add dynamic {@link SocketPermission} based on AffixSetting AUX_TRANSPORT_PORTS.
+     * If an auxiliary transport type is enabled but has no corresponding port range setting fall back to AUX_PORT_DEFAULTS.
+     *
+     * @param policy the {@link Permissions} instance to apply the dynamic {@link SocketPermission}s to.
+     * @param settings the {@link Settings} instance to read the gRPC settings from
+     */
+    private static void addSocketPermissionForAux(final Permissions policy, final Settings settings) {
+        Set<PortsRange> portsRanges = new HashSet<>();
+        for (String auxType : AUX_TRANSPORT_TYPES_SETTING.get(settings)) {
+            Setting<PortsRange> auxTypePortSettings = AUX_TRANSPORT_PORTS.getConcreteSettingForNamespace(auxType);
```

Comment:
Just a thought here, so AUX_TRANSPORT_PORTS is covering a list of aux transports, but represents a single range. Shouldn't it be a list of ranges? Fe. `9400-9401,9900-9960` ?

Reply:
AUX_TRANSPORT_PORTS is an `AffixSetting` and creates a new setting for each `aux.transport.<type>.ports` setting with distinct `type`. I thought this would make more sense than something like:
```
setting 'aux.transport.types', ['transport-1', 'transport-2', ...]
setting 'aux.transport.port-ranges', ['range-1', 'range-2']
```
As it's more clear which range belongs to which transport and there's no need to deal with mismatched list sizes.

Reply:
Totally, I missed the `AffixSetting` part of it, thank you